mmdet/models/losses/iou_loss.py

```diff
@@ -237,6 +237,78 @@ def ciou_loss(pred, target, eps=1e-7):
     return loss
 
 
+@mmcv.jit(derivate=True, coderize=True)
+@weighted_loss
+def alphaiou_loss(pred, target, alpha=3, eps=1e-7, mode='iou'):
+    # iou mode
+    mode = mode.lower()
+    assert mode in ('iou', 'ciou', 'giou', 'diou')
+
+    # overlap
+    lt = torch.max(pred[:, :2], target[:, :2])
+    rb = torch.min(pred[:, 2:], target[:, 2:])
+    wh = (rb - lt).clamp(min=0)
+    overlap = wh[:, 0] * wh[:, 1]
+
+    # union
+    ap = (pred[:, 2] - pred[:, 0]) * (pred[:, 3] - pred[:, 1])
+    ag = (target[:, 2] - target[:, 0]) * (target[:, 3] - target[:, 1])
+    union = ap + ag - overlap + eps
+
+    # change conventional iou to alpha pow
+    ious = torch.pow(overlap / union + eps, alpha)
+
+    # calculate alpha-iou according mode
+    if mode == 'iou':
+        return ious
+    else:
+        # enclose area
+        enclose_x1y1 = torch.min(pred[:, :2], target[:, :2])
+        enclose_x2y2 = torch.max(pred[:, 2:], target[:, 2:])
+        enclose_wh = (enclose_x2y2 - enclose_x1y1).clamp(min=0)
+
+        cw = enclose_wh[:, 0]
+        ch = enclose_wh[:, 1]
+
+        if mode == 'giou':
+            c_area = torch.max(cw * ch + eps, union)
+            gious = ious - torch.pow((c_area - union) / c_area + eps, alpha)
+            loss = 1 - gious
+            return loss
+
+        c2 = (cw**2 + ch**2)**alpha + eps
+
+        b1_x1, b1_y1 = pred[:, 0], pred[:, 1]
+        b1_x2, b1_y2 = pred[:, 2], pred[:, 3]
+        b2_x1, b2_y1 = target[:, 0], target[:, 1]
+        b2_x2, b2_y2 = target[:, 2], target[:, 3]
+
+        left = ((b2_x1 + b2_x2) - (b1_x1 + b1_x2))**2 / 4
+        right = ((b2_y1 + b2_y2) - (b1_y1 + b1_y2))**2 / 4
+        rho2 = (left + right)**alpha
+
+        # DIoU
+        if mode == 'diou':
+            dious = ious - rho2 / c2
+            loss = 1 - dious
+            return loss
+        else:
+            w1, h1 = b1_x2 - b1_x1, b1_y2 - b1_y1 + eps
+            w2, h2 = b2_x2 - b2_x1, b2_y2 - b2_y1 + eps
+
+            factor = 4 / math.pi**2
+            v = factor * torch.pow(
+                torch.atan(w2 / h2) - torch.atan(w1 / h1), 2)
+
+            with torch.no_grad():
+                alpha_ciou = (ious > 0.5).float() * v / (1 - ious + v)
+
+            # CIoU
```

Comment:
This comment can be put at line 296 to improve readability of the code

Reply:
> This comment can be put at line 296 to improve readability of the code

Okay, I will change it later. Thanks for your advice. :)
It's just a coarse implementation and will be improved it later.